src/utils/source.js

```diff
@@ -169,6 +169,54 @@ export function getTruncatedFileName(source: Source, length: number = 30) {
   return truncateMiddleText(getFilename(source), length);
 }
 
+/* Show a source url's unique filename for editor tabs, breakpoints, etc.
+ * Pass the url of file, and the list of file urls
+ *
+ * @memberof utils/source
+ * @static
+ */
+
+export function getUniqueFileName(mySource: Source, sources: Source[]) {
+  const myFileName = getFilename(mySource);
+
+  let myPathSegments;
+  const openEditorPathSegmentsWithSameFilename = [];
+  for (const source of sources) {
+    if (!source.url) {
+      continue;
+    }
+    const fileName = getFilename(source);
+    if (fileName === myFileName) {
+      const idx = source.url.lastIndexOf("/");
+      const pathSegments = source.url.slice(0, idx).split("/");
+      openEditorPathSegmentsWithSameFilename.push(pathSegments);
+      if (source.url === mySource.url) {
+        myPathSegments = pathSegments;
+      }
+    }
+  }
+
+  if (!myPathSegments || openEditorPathSegmentsWithSameFilename.length === 1) {
+    return myFileName;
+  }
+
+  let commonPathSegmentCount;
+  for (let i = 0, { length } = myPathSegments; i < length; i++) {
+    const myPathSegment = myPathSegments[i];
+    if (
+      openEditorPathSegmentsWithSameFilename.some(
+        segments => segments.length === i + 1 || segments[i] !== myPathSegment
+      )
+    ) {
+      commonPathSegmentCount = i;
+      break;
+    }
+  }
+  return `${myFileName} \u2014 ${[
+    ...myPathSegments.slice(commonPathSegmentCount + 1)
+  ].join("/")}`;
+}
```

Comment:
maybe we have this just return the common paths so on the component level we can do `${myFileName} \u2014  ${commonPaths}`

Reply:
nice idea. Also any nice name for the function, currently I named it 'getPath'